src/main/java/com/javadiscord/javabot/commands/user_commands/Regex.java

```diff
@@ -0,0 +1,42 @@
+package com.javadiscord.javabot.commands.user_commands;
+
+import com.javadiscord.javabot.commands.ResponseException;
+import com.javadiscord.javabot.commands.SlashCommandHandler;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.requests.restaction.interactions.ReplyAction;
+
+import java.util.regex.Pattern;
+
+public class Regex implements SlashCommandHandler {
+
+    @Override
+    public ReplyAction handle(SlashCommandEvent event) throws ResponseException {
+
+        Pattern pattern = Pattern.compile(event.getOption("regex").getAsString());
+        String string = event.getOption("string").getAsString();
+
+        if (pattern.matcher(string).matches()){
+            return event.replyEmbeds(embedBuilder(true, pattern, string).build());
+        } else {
+            return event.replyEmbeds(embedBuilder(false, pattern, string).build());
+        }
+    }
+
+    private EmbedBuilder embedBuilder(boolean matches, Pattern pattern, String string){
```

Comment:
maybe rename this to something like `buildRegexEmbed` or `buildEmbed` to match the rest of the code
 https://github.com/Java-Discord/JavaBot/blob/e63445021e39b41ce961ef83163f38b7dbc25da4/src/main/java/com/javadiscord/javabot/service/qotw/QOTWJob.java#L50

 https://github.com/Java-Discord/JavaBot/blob/main/src/main/java/com/javadiscord/javabot/commands/staff_commands/Ban.java#L60